Lib/uuid.py

```diff
@@ -673,12 +680,25 @@ def _random_getnode():
     return random.getrandbits(48) | (1 << 40)
 
 
-_node = None
-
-_NODE_GETTERS_WIN32 = [_windll_getnode, _netbios_getnode, _ipconfig_getnode]
+if _LINUX:
+    _OS_GETTERS = [_ifconfig_getnode, _ip_getnode, _arp_getnode, _lanscan_getnode]
```

Comment:
Sorry, I don't follow why linux and darwin lists are shorter than was before.
Please add a comment in code if the change is necessary.
The set of getters is a very delicate matter, would be nice to understand by reading source code comments why these checkers are selected and why others are not.

Also, please keep elements order the same as they were in `_NODE_GETTERS_UNIX` if you have no strong reason to change this.

Reply:
Short answer - yes, there are reasons for all the changes.

As to multiple lists - I have tried, manually, on these platforms - and they are the only programs that actually exist, or provide MACADDR suitable data.

as to _NODE as a name-prefix - perhaps I am missing historical context so I do not see the value of _NODE - and just trying to shorten to their "_GETTER_something character.

I changed the order because, e.g., for AIX - the old order had the only one that actually worked - as last. As the unix_getnode should work on all (posix) platforms - the others order should be irrelevant (especially as they are most likely using _uuid, not uuid.* for resolution). However, the order is to order the programs by "platform" preferred programs.

And, as to order, for linux - rather than "ifconfig" being first, it should probably be "ip" these days.

Anyway, I hear you - and should I consider additional documentation in the code, or in the "What's new/changed" documentation? I would 'guess' in the code - as these changes are meant to be transparent (and _named_something variables).

Tomorrow :)

Reply:
I'm ok with `_NODE` -> `_GETTER_*` renaming, it is an internal name.

Yes, I'm asking for documenting your findings in the code as comments: why getters for every platform are organized as you did?
I very respect your work and want to don't miss it when some developer will change these lines.
Code comments are much more visible than release notes for editors.

Reply:
OK - will work on the documentation of the following:

NF: Not Found
AV: Available
AIX:
ifconfig: no MAC; ip: NF; arp: no MAC; lanscan: NF; netstat: buggy
only netstat can return an MAC value (the parser is broken, uses '.' as divider, not always 17 characters, not on same line as keyword)

Darwin x015.fritz.box 18.6.0 Darwin Kernel Version 18.6.0:
ifconfig: ok, ip: NF; arp: ok, lanscan: NF, netstat: ok

Centos (aka RHEL):
ifconfig: RHEL - NF (default), works if available; ip: ok; arp: no MAC, even if available; lanscan: NF; netstat: no MAC;

Debian
ifconfig: ok; ip: ok; arp: no MAC; lanscan: NF; netstat: no MAC;

So, when the platform is unknown - just use the long list - hopefully one will work.

The purpose of this is to test only interfaces that are able to work (when available), and skip them otherwise - so platform based skipIf or SkipUnless statements are not needed in test_uuid.py.

A side-effect is that getnode() will never attempt to call programs known to not work.

Reply:
Good to know, thanks